src/org/exist/protocolhandler/embedded/EmbeddedInputStream.java

```diff
@@ -33,66 +33,66 @@
 
 import org.exist.protocolhandler.xmldb.XmldbURL;
 import org.exist.storage.BrokerPool;
-import org.exist.storage.io.BlockingInputStream;
-import org.exist.storage.io.BlockingOutputStream;
 
 /**
- * Read document from embedded database as a (input)stream.
+ * Read document from embedded database as an InputStream
  *
  * @author Dannes Wessels
  */
 public class EmbeddedInputStream extends InputStream {
     
-    private final static Logger logger = LogManager.getLogger(EmbeddedInputStream.class);
+    private static final Logger LOG = LogManager.getLogger(EmbeddedInputStream.class);
     
-    private PipedInputStream  bis;
-    private PipedOutputStream bos;
-    private EmbeddedDownloadThread rt;
+    private final PipedInputStream  bis;
+    private final PipedOutputStream bos;
+    private final EmbeddedDownloadThread rt;
     
     /**
      *  Constructor of EmbeddedInputStream. 
      * 
      * @param xmldbURL Location of document in database.
      * @throws MalformedURLException Thrown for illegalillegal URLs.
      */
-    public EmbeddedInputStream(XmldbURL xmldbURL) throws IOException {
-        
+    public EmbeddedInputStream(final XmldbURL xmldbURL) throws IOException {
         this(null, xmldbURL);
     }
 
-        /**
+    /**
      *  Constructor of EmbeddedInputStream.
      *
      * @param xmldbURL Location of document in database.
      * @throws MalformedURLException Thrown for illegalillegal URLs.
      */
-    public EmbeddedInputStream(BrokerPool brokerPool, XmldbURL xmldbURL) throws IOException {
+    public EmbeddedInputStream(final BrokerPool brokerPool, final XmldbURL xmldbURL) throws IOException {
 
-        logger.debug("Initializing EmbeddedInputStream");
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("Initializing EmbeddedInputStream");
+        }
 
-        bis = new PipedInputStream(2048);
-        bos = new PipedOutputStream(bis);
+        this.bis = new PipedInputStream(4096);
```

Comment:
Its an intention thing. When reading code, from only looking at the line:

```java
this.bis = new PipedInputStream(4096);
```

I can tell immediately that this line mutates shared state, i.e. it modifies a class member and not a local variable. Because the intention is clear, I don't have to search around to find out if `bis` is mutable shared state or not.